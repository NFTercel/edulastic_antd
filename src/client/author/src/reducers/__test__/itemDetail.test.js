import itemDetailReducer from "../itemDetail";

import {
  RECEIVE_ITEM_DETAIL_REQUEST,
  RECEIVE_ITEM_DETAIL_SUCCESS,
  RECEIVE_ITEM_DETAIL_ERROR,
  UPDATE_ITEM_DETAIL_REQUEST,
  UPDATE_ITEM_DETAIL_SUCCESS,
  UPDATE_ITEM_DETAIL_ERROR,
  SET_ITEM_DETAIL_DATA,
  UPDATE_ITEM_DETAIL_DIMENSION,
  SET_DRAGGING,
  DELETE_ITEM_DETAIL_WIDGET,
  UPDATE_TAB_TITLE,
  USE_TABS,
  MOVE_WIDGET
} from "../../constants/actions";

describe("item detail", () => {
  const initialState = {
    item: {
      rows: [
        {
          widgets: [],
          dimension: "100%",
          tabs: []
        }
      ]
    },
    error: null,
    loading: false,
    updating: false,
    updateError: null,
    dragging: false
  };

  it("should return the initial state", () => {
    expect(itemDetailReducer(initialState, {})).toMatchSnapshot();
  });

  it("should return the receive item detail request state", () => {
    expect(
      itemDetailReducer(initialState, {
        type: RECEIVE_ITEM_DETAIL_SUCCESS,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it("should return the receive item detail success state", () => {
    expect(
      itemDetailReducer(initialState, {
        type: RECEIVE_ITEM_DETAIL_REQUEST,
        payload: {
          item: {
            rows: [
              {
                tabs: [],
                dimension: "100%",
                widgets: [
                  {
                    widgetType: "question",
                    type: "multipleChoice",
                    title: "Multiple choice",
                    reference: "5c0f0c9ed8c7f4453f5bb51f",
                    tabIndex: 0
                  }
                ]
              }
            ],
            columns: []
          }
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the receive item detail error state", () => {
    expect(
      itemDetailReducer(initialState, {
        type: RECEIVE_ITEM_DETAIL_ERROR,
        payload: {
          error: "receive item detail error"
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the update item detail request state", () => {
    expect(
      itemDetailReducer(initialState, {
        type: UPDATE_ITEM_DETAIL_REQUEST,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it("should return the update item detail success state", () => {
    expect(
      itemDetailReducer(initialState, {
        type: UPDATE_ITEM_DETAIL_SUCCESS,
        payload: {
          item: {
            rows: [
              {
                tabs: [],
                dimension: "100%",
                widgets: [
                  {
                    widgetType: "question",
                    type: "multipleChoice",
                    title: "Multiple choice",
                    reference: "5c0f0c9ed8c7f4453f5bb51f",
                    tabIndex: 0
                  }
                ]
              }
            ],
            columns: []
          }
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the update item detail error state", () => {
    expect(
      itemDetailReducer(initialState, {
        type: UPDATE_ITEM_DETAIL_ERROR,
        payload: {
          error: "update item detail error"
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the set item detail data state", () => {
    expect(
      itemDetailReducer(initialState, {
        type: SET_ITEM_DETAIL_DATA,
        payload: {
          item: {
            rows: [
              {
                tabs: [],
                dimension: "100%",
                widgets: [
                  {
                    widgetType: "question",
                    type: "multipleChoice",
                    title: "Multiple choice",
                    reference: "5c0f0c9ed8c7f4453f5bb51f",
                    tabIndex: 0
                  }
                ]
              }
            ],
            columns: []
          }
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the update item detail dimension state", () => {
    expect(
      itemDetailReducer(initialState, {
        type: UPDATE_ITEM_DETAIL_DIMENSION,
        payload: {
          left: 10,
          right: 20
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the set dragging state", () => {
    expect(
      itemDetailReducer(initialState, {
        type: SET_DRAGGING,
        payload: {
          dragging: 5
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the delete item detail widget state", () => {
    expect(
      itemDetailReducer(initialState, {
        type: DELETE_ITEM_DETAIL_WIDGET,
        payload: {
          rowIndex: 0,
          widgetIndex: 5
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the delete item detail widget state", () => {
    expect(
      itemDetailReducer(initialState, {
        type: UPDATE_TAB_TITLE,
        payload: {
          rowIndex: 0,
          tabIndex: 2,
          value: "tab1"
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the use tabs state", () => {
    expect(
      itemDetailReducer(initialState, {
        type: USE_TABS,
        payload: {
          rowIndex: 3,
          isUseTabs: true
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the move widget state", () => {
    expect(
      itemDetailReducer(initialState, {
        type: MOVE_WIDGET,
        payload: {
          from: 0,
          to: 1
        }
      })
    ).toMatchSnapshot();
  });
});
