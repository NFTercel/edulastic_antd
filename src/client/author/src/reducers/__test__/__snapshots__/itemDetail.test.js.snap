// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`item detail should return the delete item detail widget state 1`] = `
Object {
  "dragging": false,
  "error": null,
  "item": Object {
    "rows": Array [
      Object {
        "dimension": "100%",
        "tabs": Array [],
        "widgets": Array [],
      },
    ],
  },
  "loading": false,
  "updateError": null,
  "updating": false,
}
`;

exports[`item detail should return the delete item detail widget state 2`] = `
Object {
  "dragging": false,
  "error": null,
  "item": Object {
    "rows": Array [
      Object {
        "dimension": "100%",
        "tabs": Array [
          undefined,
          undefined,
          "tab1",
        ],
        "widgets": Array [],
      },
    ],
  },
  "loading": false,
  "updateError": null,
  "updating": false,
}
`;

exports[`item detail should return the initial state 1`] = `
Object {
  "dragging": false,
  "error": null,
  "item": Object {
    "rows": Array [
      Object {
        "dimension": "100%",
        "tabs": Array [],
        "widgets": Array [],
      },
    ],
  },
  "loading": false,
  "updateError": null,
  "updating": false,
}
`;

exports[`item detail should return the receive item detail error state 1`] = `
Object {
  "dragging": false,
  "error": "receive item detail error",
  "item": Object {
    "rows": Array [
      Object {
        "dimension": "100%",
        "tabs": Array [],
        "widgets": Array [],
      },
    ],
  },
  "loading": false,
  "updateError": null,
  "updating": false,
}
`;

exports[`item detail should return the receive item detail request state 1`] = `
Object {
  "dragging": false,
  "error": null,
  "item": undefined,
  "loading": false,
  "updateError": null,
  "updating": false,
}
`;

exports[`item detail should return the receive item detail success state 1`] = `
Object {
  "dragging": false,
  "error": null,
  "item": Object {
    "rows": Array [
      Object {
        "dimension": "100%",
        "tabs": Array [],
        "widgets": Array [],
      },
    ],
  },
  "loading": true,
  "updateError": null,
  "updating": false,
}
`;

exports[`item detail should return the set dragging state 1`] = `
Object {
  "dragging": 5,
  "error": null,
  "item": Object {
    "rows": Array [
      Object {
        "dimension": "100%",
        "tabs": Array [],
        "widgets": Array [],
      },
    ],
  },
  "loading": false,
  "updateError": null,
  "updating": false,
}
`;

exports[`item detail should return the set item detail data state 1`] = `
Object {
  "dragging": false,
  "error": null,
  "item": Object {
    "columns": Array [],
    "rows": Array [
      Object {
        "dimension": "100%",
        "tabs": Array [],
        "widgets": Array [
          Object {
            "reference": "5c0f0c9ed8c7f4453f5bb51f",
            "tabIndex": 0,
            "title": "Multiple choice",
            "type": "multipleChoice",
            "widgetType": "question",
          },
        ],
      },
    ],
  },
  "loading": false,
  "updateError": null,
  "updating": false,
}
`;

exports[`item detail should return the update item detail dimension state 1`] = `
Object {
  "dragging": false,
  "error": null,
  "item": Object {
    "rows": Array [
      Object {
        "dimension": 10,
        "tabs": Array [],
        "widgets": Array [],
      },
      Object {
        "dimension": 20,
        "tabs": Array [
          "Tab 1",
          "Tab 2",
        ],
        "widgets": Array [],
      },
    ],
  },
  "loading": false,
  "updateError": null,
  "updating": false,
}
`;

exports[`item detail should return the update item detail error state 1`] = `
Object {
  "dragging": false,
  "error": null,
  "item": Object {
    "rows": Array [
      Object {
        "dimension": "100%",
        "tabs": Array [],
        "widgets": Array [],
      },
    ],
  },
  "loading": false,
  "updateError": "update item detail error",
  "updating": false,
}
`;

exports[`item detail should return the update item detail request state 1`] = `
Object {
  "dragging": false,
  "error": null,
  "item": Object {
    "rows": Array [
      Object {
        "dimension": "100%",
        "tabs": Array [],
        "widgets": Array [],
      },
    ],
  },
  "loading": false,
  "updateError": null,
  "updating": true,
}
`;

exports[`item detail should return the update item detail success state 1`] = `
Object {
  "dragging": false,
  "error": null,
  "item": Object {
    "columns": Array [],
    "rows": Array [
      Object {
        "dimension": "100%",
        "tabs": Array [],
        "widgets": Array [
          Object {
            "reference": "5c0f0c9ed8c7f4453f5bb51f",
            "tabIndex": 0,
            "title": "Multiple choice",
            "type": "multipleChoice",
            "widgetType": "question",
          },
        ],
      },
    ],
  },
  "loading": false,
  "updateError": null,
  "updating": false,
}
`;
